lumicks/pylake/force_calibration/calibration_models.py

```diff
@@ -99,3 +102,135 @@ def calibration_results(self, fc, diffusion_constant):
+           Calibration of optical tweezers with positional detection in the back focal plane,
+           Review of scientific instruments 77, 103101 (2006).
+    """
+
+    def __name__(self):
+        return "ActiveCalibrationModel"
+
+    def __init__(
+        self,
+        driving_data,
+        force_voltage_data,
+        sample_rate,
+        bead_diameter,
+        driving_frequency_guess,
+        viscosity=1.002e-3,
+        temperature=20,
+        num_windows=5,
+    ):
+        """
+        Active Calibration Model.
+
+        This model fits data acquired in the presence of nanostage or mirror oscillation.
+
+        Parameters
+        ----------
+        driving_data : array_like
+            Array of driving data.
+        force_voltage_data : array_like
+            Uncalibrated force data in volts.
+        sample_rate : float
+            Sample rate at which the signals we acquired.
+        bead_diameter : float
+            Bead diameter [um].
+        driving_frequency_guess : float
+            Guess of the driving frequency.
+        viscosity : float, optional
+            Liquid viscosity [Pa*s].
+        temperature : float, optional
+            Liquid temperature [Celsius].
+        num_windows : int, optional
+            Number of windows to average for the uncalibrated force. Using a larger number of
+            windows potentially increases the bleed, but may be useful when the SNR is low.
+        """
+        super().__init__(bead_diameter)
+        self.driving_frequency_guess = driving_frequency_guess
+        self.sample_rate = sample_rate
+        self.viscosity = viscosity
+        self.temperature = temperature
+        self.num_windows = num_windows
+
+        # Estimate driving input and response
+        amplitude_um, self.driving_frequency = estimate_driving_input_parameters(
+            sample_rate, driving_data, driving_frequency_guess
+        )
+        self.driving_amplitude = amplitude_um * 1e-6
+        self.response_power_density, self.frequency_resolution = response_power_density(
+            force_voltage_data, sample_rate, self.driving_frequency, num_windows
+        )
+
+    def calibration_parameters(self):
+        return {
+            "Bead diameter": CalibrationParameter("Bead diameter", self.bead_diameter, "um"),
+            "Driving frequency (guess)": CalibrationParameter(
+                "Driving frequency (guess)", self.driving_frequency_guess, "Hz"
+            ),
+            "Sample rate": CalibrationParameter("Sample rate", self.sample_rate, "Hz"),
+            "Temperature": CalibrationParameter("Liquid temperature", self.temperature, "C"),
+            "Viscosity": CalibrationParameter("Liquid viscosity", self.viscosity, "Pa*s"),
+            "num_windows": CalibrationParameter("Number of averaged windows", self.num_windows, ""),
+        }
+
+    def _power_theory(self, f_corner):
```

Comment:
May I suggest: `_ideal_power_spectrum`?

Reply:
That's not what this is though.

This is absolute power expected from a spike (integral over a delta spike in the power spectrum).

There were two comments on this function, so it's clear it needs some better clarification on my end.

How about:

```python
    def _theoretical_driving_power(self, f_corner):
        """Compute the power expected for a given driving input.
        
        When driving the stage or trap, we expect to see a delta spike in the power density 
        spectrum. This function returns the expected power contribution of the bead motion to the
        power spectrum. It corresponds to the driven power spectrum minus the thermal power spectrum
        integrated over the frequency bin corresponding to the driving input."""
        return self.driving_amplitude ** 2 / (2 * (1 + (f_corner / self.driving_frequency) ** 2))
```

Reply:
This is perfectly clear now! :+1: